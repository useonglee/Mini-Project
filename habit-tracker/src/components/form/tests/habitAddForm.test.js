import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import HabitAddForm from "../habitAddForm";
import renderer from "react-test-renderer";

describe("habit add form", () => {
  it("renders", () => {
    // snapshot testing
    const component = renderer.create(<HabitAddForm onAdd={jest.fn()} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("habit submit", () => {
    let onAdd;
    let input;
    let button;

    beforeAll(() => {
      onAdd = jest.fn();
      render(<HabitAddForm onAdd={onAdd} />);
      input = screen.getByPlaceholderText("Habit");
      button = screen.getByText("Add");
    });

    it("calls onAdd when button is clicked", () => {
      userEvent.type(input, "New Habit");
      userEvent.click(button);

      expect(onAdd).toHaveBeenCalledWith("New Habit");
    });

    it("does not call onAdd when the habit is empty", () => {
      userEvent.type(input, "");
      userEvent.click(button);

      expect(onAdd).toHaveBeenCalledTimes(0);
    });
  });
});
