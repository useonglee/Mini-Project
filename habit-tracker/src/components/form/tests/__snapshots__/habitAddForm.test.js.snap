// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habit add form renders 1`] = `
<form
  className="add-form"
  onSubmit={[Function]}
>
  <input
    className="add-input"
    placeholder="Habit"
    type="text"
  />
  <button
    className="add-button"
  >
    Add
  </button>
</form>
`;
